core/io/resource.h

```diff
@@ -149,11 +149,18 @@ class Resource : public RefCounted {
 	~Resource();
 };
 
+struct EvictionListenRecord {
+	void (*listener)(void *p_context, const String &p_path);
+	void *context;
+};
+
 class ResourceCache {
 	friend class Resource;
 	friend class ResourceLoader; //need the lock
 	static Mutex lock;
 	static HashMap<String, Resource *> resources;
+	static Mutex listener_lock;
```

Comment:
```suggestion
	static Mutex listener_mutex;
```
Better naming for `Mutex` IMO, calling a `Mutex` a lock is not entirely correct (and I'd sugest renaming the mutex above)